mdata/cache/cache.go

```diff
@@ -0,0 +1,107 @@
+package cache
+
+import (
+	"flag"
+	"sync"
+
+	"github.com/raintank/metrictank/iter"
+	accnt "github.com/raintank/metrictank/mdata/cache/accnt"
+	"github.com/rakyll/globalconf"
+)
+
+var (
+	maxSize uint64
+)
+
+func init() {
+	flags := flag.NewFlagSet("chunk-cache", flag.ExitOnError)
+	// (1024 ^ 3) * 4 = 4294967296 = 4G
+	flags.Uint64Var(&maxSize, "max-size", 4294967296, "Maximum size of chunk cache in bytes")
+	globalconf.Register("chunk-cache", flags)
+}
```

Comment:
hmm... it does appear to be working. i think what happens is this:
- during init the options get registered in `globalconf` and the variables get set to the default values
- then in the `main()` function `conf.ParseAll()` is called, which reads the config file and sets the variables
- from the on the variables have the correct values from the config file

so that would be a reasonable explanation why it's ok to register this variable in the `init` of the module even before the config file is known to `globalconf`.